source/MaterialXRuntime/Private/PvtApi.h

```diff
@@ -292,6 +307,70 @@ class PvtApi
         return _implementationSearchPaths;
     }
 
+    const FileSearchPath& getMaterialXDefinitionPaths() const
+    {
+        return _materialXDefinitionPaths;
+    }
+
+    const FileSearchPath& getGlobalDefinitionPaths() const
+    {
+        return _globalDefinitionPaths;
+    }
+
+    const FileSearchPath& getLocalDefinitionPaths() const
+    {
+        return _localDefinitionPaths;
+    }
+
+    const FileSearchPath& getMaterialXCoreDefinitionPaths() const
+    {
+        return _materialXCoreDefinitionPaths;
+    }
+
+    const FileSearchPath getDefinitionPaths(bool includeSubFolders=true) const
+    {
+        FileSearchPath definitionPaths;
+        FileSearchPath materialXDefinitionPaths = getMaterialXDefinitionPaths();
+        if (!materialXDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(materialXDefinitionPaths);
+        }
+        FileSearchPath globalDefinitionPaths = getGlobalDefinitionPaths();
+        if (!globalDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(globalDefinitionPaths);
+        }
+        FileSearchPath localDefinitionPaths = getLocalDefinitionPaths();
+        if (!localDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(localDefinitionPaths);
+        }
+        FileSearchPath materialXCoreDefinitionPaths = getMaterialXCoreDefinitionPaths();
+        if (!materialXCoreDefinitionPaths.isEmpty())
+        {
+            definitionPaths.append(materialXCoreDefinitionPaths);
+        }
+
+        if (includeSubFolders)
```

Comment:
Gets the subfolders

Reply:
I'm wondering for the order of this whether we should look at core + mtlx definition paths, then local + global as this defines search order.

Reply:
Ok, that's easy to change.